### PR TITLE
12. Jenkins. As code and 13.Docker.Lading - Solomenik Mikhail homework

### PR DESCRIPTION
**Pull request by elegantsignal**
Don't pay  attention to fixes in `10.Jenkins.Start`. They need to run Docker containers from my Jenkins container.

### CODE REVIEW

File: Mikhail_Solomenik/12.Jenkins.As_code/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            }
+        }
+
+        stage('Run test container') {
+            steps {
+                script {
+                    sh '''
+docker run -it -d --rm \
+    --name spring-echo-example-run \
+    -v /home/jenkins/jenkins_home/workspace/spring-echo-example:/home/maven/app \
```
Comment (pluhin) [inline Mikhail_Solomenik/12.Jenkins.As_code/Jenkinsfile]: Workspace path should be as env variable

File: Mikhail_Solomenik/12.Jenkins.As_code/Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+                    }
+                }
+            }
+        }
+
+        stage('Build maven project') { 
+            agent {
+                docker {
+                    image "elegantsignal/spring-echo-example:latest"
+                    args '-v /home/jenkins/jenkins_home/workspace/spring-echo-example:/home/maven/app'
```
Comment (pluhin) [inline Mikhail_Solomenik/12.Jenkins.As_code/Jenkinsfile]: Workspace path should be as env variable

File: Mikhail_Solomenik/12.Jenkins.As_code/12.Jenkins.As_code.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+            }
+        }
+
+        stage('Run test container') {
+            steps {
+                script {
+                    sh '''
+docker run -it -d --rm \
+    --name spring-echo-example-run \
+    -v /home/jenkins/jenkins_home/workspace/spring-echo-example:/home/maven/app \
```
Comment (pluhin) [inline Mikhail_Solomenik/12.Jenkins.As_code/12.Jenkins.As_code.md]: Workspace path should be as env variable

File: Mikhail_Solomenik/12.Jenkins.As_code/12.Jenkins.As_code.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+                    }
+                }
+            }
+        }
+
+        stage('Build maven project') { 
+            agent {
+                docker {
+                    image "elegantsignal/spring-echo-example:latest"
+                    args '-v /home/jenkins/jenkins_home/workspace/spring-echo-example:/home/maven/app'
```
Comment (pluhin) [inline Mikhail_Solomenik/12.Jenkins.As_code/12.Jenkins.As_code.md]: Workspace path should be as env variable

**Comment by elegantsignal** [comment]
Fixed

**Comment by pluhin** [comment]
> Don't pay attention to fixes in `10.Jenkins.Start`. They need to run Docker containers from my Jenkins container.

Good :)

**Pull request merged**